File: C++/power_of_two.cpp
// no. 231

class Solution {
public:
  bool isPowerOfTwo(int n) {
    if (n <= 0) {
      return false;
    }
    if (n <= 2) {
      return true;
    }
    if (n % 2) {
      return false;
    }
    bool ans = isPowerOfTwo(n / 2);
    return ans;
  }
};

File: python/power_of_two.py
#no. 231

class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        if n <= 0:
            return False
        if n <= 2:
            return True
        if n % 2:
            return False
        ans = self.isPowerOfTwo(n / 2)
        return ans
